7_react/react1/week2/todo-app/src/todoList/TodoList.jsx

```diff
@@ -0,0 +1,53 @@
+import TimeCount from "../timer/Timer";
+import TodoItem from "../todoItem/TodoItem";
+import todos from "../todoItem/todos";
+import { useState } from "react";
+
+function AppTodo() {
+  // const data = [...todos];
+  const [todolist, setTodoList] = useState([...todos]);
+
+  const addTodo = () => {
+    const newTodo = {id:new Date(), name: "new todo", date: new Date(), status: false };
+    setTodoList((todolist) => { return [...todolist, newTodo]; });
+  }
+ 
+  const onDeleteClick = (id) => {
+    setTodoList(todolist.filter(todo => todo.id !== id));
+  }
+
+  const TodoList = (props) => {
+    return (
+      <div>
+        <button onClick={addTodo}>Add todo</button>
+        <ul>
+          {props.todos.length > 0 ? (
+            props.todos.map((todo) => {
+              return (
+                <TodoItem
```

Comment:
Always add a unique key prop when using map to render a list of items in react
`key={todo.id}`